nova_vm/src/ecmascript/abstract_operations/type_conversion.rs

```diff
@@ -471,3 +471,87 @@ pub(crate) fn to_object(_agent: &mut Agent, argument: Value) -> JsResult<Object>
         _ => Ok(Object::try_from(argument).unwrap()),
     }
 }
+
+/// ### [7.1.19 ToPropertyKey ( argument )](https://tc39.es/ecma262/#sec-topropertykey)
+pub(crate) fn to_property_key(agent: &mut Agent, argument: Value) -> JsResult<Value> {
```

Comment:
**issue:** While it's not spec-wise exactly the same, I think we might want to replace steps 2 and 3 with mostly a conversion into `PropertyKey`. There are a few special cases (Boolean, BigInts larger than 2^53 - 1, and non-integer Numbers) that should take the normal step 3 route but aside from those since this is an internal operation instead of a public API we should aim to return the type that makes most sense to the engine. And in this case I do feel that `PropertyKey` makes the most sense :)